### PR TITLE
Enhancement: Transition JCK logic in testJobTemplate into jck.mk

### PR DESCRIPTION
**Pull request by YeeSkywalker**
Signed out: Yee Lin
Date: Feb 1st, 2022
Resolves https://github.com/adoptium/aqa-tests/issues/1754

Please review when you are free! Thank you for your time in advance.

### CODE REVIEW

File: jck/jck.mk [hunk 1]
```diff
@@ -68,8 +68,13 @@ endif
 OTHER_OPTS=
 # if JDK_IMPL is openj9 or ibm
 ifneq ($(filter openj9 ibm, $(JDK_IMPL)),)
- OTHER_OPTS=-Xtrace:maximal=all{level2}
- export CONFIG_ALT_PATH:=$(JCK_ROOT)$(D)config$(D)default
```
Comment (Mesbah-Alam) [inline jck/jck.mk:72]: Please put back the deleted line 72. It should not be removed as part of this PR.

**Review by Mesbah-Alam — CHANGES_REQUESTED** [review]

File: jck/jck.mk (hunk 1 above)
Comment (YeeSkywalker) [reply jck/jck.mk:72]: Hi @Mesbah-Alam, I just revoked the deleted line, please review when you are free!

**Comment by smlambert** [comment]
@Mesbah-Alam - are you happy with this updated PR?

**Review by smlambert — COMMENTED** [review]
I think this PR also needs to remove the code from testJobTemplate file, https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/testJobTemplate#L169-L180 as we do not need it in both places.

**Comment by Mesbah-Alam** [comment]
@YeeSkywalker , yes, please remove the logic from its old location, as Shelley pointed out [above](https://github.com/adoptium/aqa-tests/pull/3309#pullrequestreview-873333121). I will then run some tck grinders with your PR on z/OS and a non-z/OS platform to ensure all is well. 

Thanks for the reminder @ShelleyLambert .

File: jck/jck.mk [hunk 2]
```diff
@@ -68,8 +68,14 @@ endif
...
+  OTHER_OPTS=" -Xtrace:maximal=all{level2} -Xfuture "
```
Comment (Mesbah-Alam) [inline jck/jck.mk:71]: Please remove the Quotes ("). They are not needed.

File: jck/jck.mk [hunk 3]
```diff
@@ -68,8 +68,14 @@ endif
...
+  export CONFIG_ALT_PATH:=$(JCK_ROOT)$(D)config$(D)default
+  ifeq ($(OS), OS/390)
+    OTHER_OPTS += " -Dcom.ibm.tools.attach.enable=yes "
+  endif
+  ifneq (8, $(JDK_VERSION))
+    OTHER_OPTS += " --enable-preview "
```
Comment (Mesbah-Alam) [inline jck/jck.mk:77]: The Quotes (") need to be removed from here too.

File: jck/jck.mk [hunk 4]
```diff
@@ -68,8 +68,14 @@ endif
 OTHER_OPTS=
 # if JDK_IMPL is openj9 or ibm
 ifneq ($(filter openj9 ibm, $(JDK_IMPL)),)
- OTHER_OPTS=-Xtrace:maximal=all{level2}
- export CONFIG_ALT_PATH:=$(JCK_ROOT)$(D)config$(D)default
+  OTHER_OPTS=" -Xtrace:maximal=all{level2} -Xfuture "
+  export CONFIG_ALT_PATH:=$(JCK_ROOT)$(D)config$(D)default
+  ifeq ($(OS), OS/390)
```
Comment (Mesbah-Alam) [inline jck/jck.mk:73]: Also, this condition doesn't detect z/OS platform correctly (checked in a grinder). I suggest we use the following: 

```
ifneq (,$(findstring zos,$(SPEC)))
```

**Review by Mesbah-Alam — CHANGES_REQUESTED** [review]

**Comment by YeeSkywalker** [comment]
> @YeeSkywalker , yes, please remove the logic from its old location, as Shelley pointed out [above](https://github.com/adoptium/aqa-tests/pull/3309#pullrequestreview-873333121). I will then run some tck grinders with your PR on z/OS and a non-z/OS platform to ensure all is well.
> 
> Thanks for the reminder @smlambert !

Hi @Mesbah-Alam, I have updated jck.mk and testJobTemplate file according to your comments, please review when you are free. Thank you for your time!

**Comment by Mesbah-Alam** [comment]
Looks good, Thanks, @YeeSkywalker ! 

PR tested on z/OS with Grinder/20802, non-z/OS with Grinder/20803.

**Review by smlambert — APPROVED** [review]

**Pull request merged**